waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMappingFactoryImpl.java

```diff
@@ -52,7 +53,8 @@ public MetaStoreMappingFactoryImpl(
 
   private CloseableThriftHiveMetastoreIface createClient(AbstractMetaStore metaStore) {
     try {
-      return metaStoreClientFactory.newInstance(metaStore);
+      MetastoreClientFactoryHelper helper = new MetastoreClientFactoryHelper(metaStore);
```

Comment:
I find it really weird that I have to create two object with the same argument `metastore`. Sure the factory can create it as well? If this is done to make it more testable then the problem is just shifted right and this method now is just as hard to test.